### PR TITLE
NSW parser

### PR DESCRIPTION
**Pull request by s3364859**
@6nop @s3364942 @s3494110 @s3482247 

In this PR:
 - Project re-targeting - some had slipped below the maximum framework version we're targeting
 - Migrations for crime and location
 - Object model types for crime and location (and associated)
 - NSW Excel XML parser migration
 - Additional repository types for associated object model types

dumbarse

### CODE REVIEW

File: CPT331.Core/CPT331.Core.csproj [hunk 1]
```diff
@@ -9,8 +9,9 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>CPT331.Core</RootNamespace>
     <AssemblyName>CPT331.Core</AssemblyName>
-    <TargetFrameworkVersion>v4.5.2</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
```
Comment (s3364859) [inline CPT331.Core/CPT331.Core.csproj]: These somehow slipped.

File: CPT331.Data.Migration/Migrations/M0004.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	{
+		public override void Down()
+		{
+			//	Nothing to see here
+		}
+
+		public override void Up()
+		{
+			//	Obviously this isn't going to work for you if you don't have the file (which is ~50 MB atm)
+			NswXmlParser nswXmlParser = new NswXmlParser(@"D:\Downloads\Crime Data\NSW\Book1.xml");
```
Comment (s3364859) [inline CPT331.Data.Migration/Migrations/M0004.cs]: Relies on an enormous XML file atm.

File: CPT331.Data.Parsers/NswXmlParser.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+						offence = offences.Where(m => (m.Name == suboffenceName)).FirstOrDefault();
+					}
+
+					DateTime dateTime = new DateTime(year, 1, 1);
+
+					for (int i = 3; i < xmlNode.ChildNodes.Count; i++)
+					{
+						int count = Convert.ToInt32(xmlNode.ChildNodes[i].InnerText);
+
+						CrimeRepository.AddCrime(count, localGovernmentArea.ID, dateTime.Month, offence.ID, dateTime.Year);
```
Comment (s3364859) [inline CPT331.Data.Parsers/NswXmlParser.cs]: To do: replace multiple repository calls with one scripted transaction.

File: CPT331.Data/CrimeRepository.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+			}
+
+			return id;
+		}
+
+		public static Crime GetCrimeByID(int id)
+		{
+			Crime crime = null;
+
+			//	using (SqlConnection sqlConnection = SqlConnectionFactory.NewSqlConnetion())
```
Comment (s3364859) [inline CPT331.Data/CrimeRepository.cs]: TODO

File: CPT331.Data/CrimeRepository.cs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+			//	}
+
+			return crime;
+		}
+
+		public static List<Crime> GetCrimesByCoordinate(double latitude, double longitude, double radius)
+		{
+			List<Crime> crimes = null;
+
+			//	using (SqlConnection sqlConnection = SqlConnectionFactory.NewSqlConnetion())
```
Comment (s3364859) [inline CPT331.Data/CrimeRepository.cs]: TODO

**Comment by 6nop** [comment]
Updates look good Carl 👍

**Pull request merged**